Interop/StubGenerator/src/test/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrToMetadataTests.kt

```diff
@@ -0,0 +1,128 @@
+        stubs.accept(it, null)
+        it.writeModule()
+    }
+
+    private fun checkUniqIdPresence(metadata: KmModuleFragment) {
+        metadata.classes.forEach { assertNotNull(it.uniqId) }
+        metadata.pkg?.let { pkg ->
+            pkg.functions.forEach { assertNotNull(it.uniqId) }
+            pkg.properties.forEach { assertNotNull(it.uniqId) }
+            pkg.typeAliases.forEach { assertNotNull(it.uniqId) }
+        }
+    }
+
+    @Test
+    fun `single simple function`() {
+        val packageName = "single_function"
+        val function = createTrivialFunction("hello")
+        val metadata = createMetadata(packageName, functions = listOf(function))
+        with (metadata) {
+            checkUniqIdPresence(this)
+            assertEquals(packageName, packageName)
+            assertTrue(classes.isEmpty())
+            assertNotNull(pkg)
+            assertTrue(pkg!!.functions.size == 1)
+
+            val kmFunction = pkg!!.functions[0]
+            assertEquals(kmFunction.name, function.name)
+            assertEquals(0, kmFunction.valueParameters.size)
+            val returnTypeClassifier = kmFunction.returnType.classifier
+            assertTrue(returnTypeClassifier is KmClassifier.Class)
+            assertEquals("kotlin/Int", returnTypeClassifier.name)
+        }
+    }
+
+    @Test
+    fun `single constant`() {
+        val property = createTrivialIntegerConstantProperty("meaning", 42)
+        val metadata = createMetadata("single_property", properties = listOf(property))
+        with (metadata) {
+            checkUniqIdPresence(this)
+            assertNotNull(pkg)
+            assertTrue(pkg!!.properties.size == 1)
+
+            val kmProperty = pkg!!.properties[0]
+            assertEquals(kmProperty.name, property.name)
+
+            val compileTimeValue = kmProperty.compileTimeValue
+            assertNotNull(compileTimeValue)
+            assertTrue(compileTimeValue is KmAnnotationArgument.IntValue)
+            assertEquals(42, compileTimeValue.value)
+        }
+    }
+
+    @Test
+    fun `single simple function after metadata deserialization`() {
+        val packageName = "single_function"
+        val function = createTrivialFunction("hello")
+        val metadata = createMetadata(packageName, functions = listOf(function))
+
+        with (metadata) {
+            checkUniqIdPresence(this)
+            assertEquals(packageName, packageName)
+            assertTrue(classes.isEmpty())
+            assertNotNull(pkg)
+            assertTrue(pkg!!.functions.size == 1)
+
+            val kmFunction = pkg!!.functions[0]
+            assertEquals(kmFunction.name, function.name)
+            assertEquals(0, kmFunction.valueParameters.size)
+            val returnTypeClassifier = kmFunction.returnType.classifier
+            assertTrue(returnTypeClassifier is KmClassifier.Class)
+            assertEquals("kotlin/Int", returnTypeClassifier.name)
+        }
+    }
```

Comment:
How is it different from `single simple function`?

Reply:
Oops. Forgot to remove.